lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Transaction.kt

```diff
@@ -144,7 +143,24 @@ data class DeclareTransaction(
     override val nonce: Felt,
 
     override val type: TransactionType = TransactionType.DECLARE,
-) : Transaction()
+
+    private val contractDefinition: ContractDefinition? = null,
+) : Transaction() {
+    @Throws(ConvertingToPayloadFailedException::class)
```

Comment:
I think we have toPayload in other places as well. I think we can resuse it there as well?

Reply:
I'll try to think of a way of making `toPayload` more defined and reusable.

Reply:
I don't think this comment still applies?